File: apple/Sources/ReactNativeWatchOS/Modifiers/ButtonStyle.swift
import SwiftUI

/// SwiftUI `.buttonStyle(_:)`. Maps the `style` string to a primitive
/// button style. Unknown values fall back to `.automatic`.
enum RNWButtonStyleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("buttonStyle") { view, params, _ in
            switch params.string("style") {
            case "bordered":
                return AnyView(view.buttonStyle(.bordered))
            case "borderedProminent":
                return AnyView(view.buttonStyle(.borderedProminent))
            case "borderless":
                return AnyView(view.buttonStyle(.borderless))
            case "plain":
                return AnyView(view.buttonStyle(.plain))
            default:
                return AnyView(view.buttonStyle(.automatic))
            }
        }
    }
}

File: src/modifiers/buttonStyle.ts
import { createModifier } from './createModifier';

export type ButtonStyle =
  | 'automatic'
  | 'bordered'
  | 'borderedProminent'
  | 'borderless'
  | 'plain';

/// SwiftUI `.buttonStyle(_:)`. Sets the visual style applied to `Button`s
/// within the view. `'automatic'` defers to the platform default.
export function buttonStyle(style: ButtonStyle = 'automatic') {
  return createModifier('buttonStyle', { style });
}
